src/dotnet/APIView/APIViewWeb/LeanControllers/SamplesRevisionsController.cs

```diff
@@ -0,0 +1,125 @@
+        {
+            if (!string.IsNullOrEmpty(usageSampleAPIParam.Content))
+            {
+                var samplesRevision = await _samplesRevisionsManager.UpsertSamplesRevisionsAsync(User, reviewId, usageSampleAPIParam.Content, usageSampleAPIParam.Title);
+                return new LeanJsonResult(samplesRevision, StatusCodes.Status200OK);
+            } else if (usageSampleAPIParam.File != null)
+            {
+                var samplesRevision = await _samplesRevisionsManager.UpsertSamplesRevisionsAsync(User, reviewId, usageSampleAPIParam.File.OpenReadStream(), usageSampleAPIParam.Title, usageSampleAPIParam.File.FileName);
+                return new LeanJsonResult(samplesRevision, StatusCodes.Status200OK);
+            }
+            else
+            {
+                return BadRequest();
+            }
+        }
+
+        /// <summary>
+        /// Update usage sample revision
+        /// </summary>
+        /// <param name="reviewId"></param>
+        /// <param name="sampleRevisionId"></param>
+        /// <param name="usageSampleAPIParam"></param>
+        /// <returns></returns>
+        [HttpPatch("{reviewId}/update", Name = "UpdateUsageSample")]
+        public async Task UpdateUsageSampleAsync(string reviewId, string sampleRevisionId, [FromForm] UsageSampleAPIParam usageSampleAPIParam)
+        {
+            await _samplesRevisionsManager.UpdateSamplesRevisionAsync(User, reviewId, sampleRevisionId, usageSampleAPIParam.Content, usageSampleAPIParam.Title);;
+        }
+
+        /// <summary>
+        /// Endpoint used by Client SPA for Deleting Usage Sample.
+        /// </summary>
+        /// <param name="deleteParams"></param>
+        /// <returns></returns>
+        [HttpPut("delete", Name = "DeleteUsageSample")]
+        public async Task DeleteUsageSampleAsync([FromBody] SamplesRevisionSoftDeleteParam deleteParams)
+        {
+            foreach (var revisionId in deleteParams.samplesRevisionIds)
+            {
+                await _samplesRevisionsManager.DeleteSamplesRevisionAsync(User, deleteParams.reviewId, revisionId);
+            }
+        }
+
+        [Route("{reviewId}/content")]
+        [HttpGet]
+        public async Task<ActionResult<string>> GetSamplesContentAsync(string reviewId, [FromQuery] string activeSamplesRevisionId)
+        {
+            var activeSamplesRevision = await _samplesRevisionsManager.GetSamplesRevisionAsync(reviewId, activeSamplesRevisionId);
+
+            if (activeSamplesRevision != null)
+            {
+
+                string samplesContent = await _samplesRevisionsManager.GetSamplesRevisionContentAsync(activeSamplesRevision.OriginalFileId);
+
+                return new LeanJsonResult(samplesContent, StatusCodes.Status200OK);
+            }
+
+            return new LeanJsonResult("Invalid SamplesRevision", StatusCodes.Status500InternalServerError);
```

Comment:
I think we should return 404 instead of 500 error code here when sample is missing. Service side error 500 will be misleading.